ami/main/api/views.py

```
@@ -706,6 +711,81 @@ def unstar(self, _request, pk=None) -> Response:
+            thumb = obj.thumbnails.get(**size)
+        except SourceImageThumbnail.DoesNotExist:
+            thumb = None
+        if not thumb or not default_storage.exists(thumb.path):
+            if obj.path and obj.deployment and obj.deployment.data_source:
+                config = obj.deployment.data_source.config
+                # Get the file
+                try:
+                    img = ami.utils.s3.read_image(config=config, key=obj.path)
+                except ami.utils.s3.botocore.exceptions.ClientError as e:
+                    logger.error(f"Could not read image for {obj.path}: {e}")
+                    raise api_exceptions.NotFound(detail=f"SourceImage with id {obj.id} media not found")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)